### PR TITLE
chore(cardano): remove legacy types

### PR DESCRIPTION
**Pull request by davidmisiak**

### CODE REVIEW

File: common/protob/messages-cardano.proto [hunk 1]
```diff
@@ -287,18 +287,18 @@ message CardanoPoolMetadataType {
...
-    required uint32 owners_count = 11;                                   // number of pool owners
-    required uint32 relays_count = 12;                                   // number of pool relays
+    required bytes pool_id = 1;                         // stake pool cold public key hash (28 bytes)
+    required bytes vrf_key_hash = 2;                    // VRF key hash (32 bytes)
+    required uint64 pledge = 3;                         // pledge amount in lovelace
+    required uint64 cost = 4;                           // cost in lovelace
+    required uint64 margin_numerator = 5;               // pool margin numerator
+    required uint64 margin_denominator = 6;             // pool margin denominator
+    required string reward_account = 7;                 // bech32 reward address where the pool receives rewards
+    // repeated CardanoPoolOwner owners = 8;            // legacy pool owners list, removed
```
Comment (gabrielKerekes) [inline common/protob/messages-cardano.proto:297]:
```suggestion
    // repeated CardanoPoolOwner owners = 8;            // legacy pool owners list - support for pre-tx-streaming firmwares dropped
```

File: common/protob/messages-cardano.proto [hunk 2]
```diff
@@ -287,18 +287,18 @@ message CardanoPoolMetadataType {
...
+    // repeated CardanoPoolRelayParameters relays = 9;  // legacy pool relays list, removed
```
Comment (gabrielKerekes) [inline common/protob/messages-cardano.proto:298]:
```suggestion
    // repeated CardanoPoolRelayParameters relays = 9;  // legacy pool relays list - support for pre-tx-streaming firmwares dropped
```

**Review by gabrielKerekes — APPROVED** [review]
One tiny suggestions. Otherwise LGTM.

**Comment by davidmisiak** [comment]
Merged here https://github.com/trezor/trezor-firmware/pull/2228

**Pull request closed without merging**